cmd/limactl/disk.go

```diff
@@ -0,0 +1,305 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"text/tabwriter"
+
+	"github.com/docker/go-units"
+	"github.com/lima-vm/lima/pkg/qemu"
+	"github.com/lima-vm/lima/pkg/store"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newDiskCommand() *cobra.Command {
+	var diskCommand = &cobra.Command{
+		Use:   "disk",
+		Short: "Lima disk management",
+		Example: `  Create a disk:
+  $ limactl disk create DISK --size SIZE
+
+  List existing disks:
+  $ limactl disk ls
+
+  Delete a disk:
+  $ limactl disk delete DISK`,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+	diskCommand.AddCommand(
+		newDiskCreateCommand(),
+		newDiskListCommand(),
+		newDiskDeleteCommand(),
+		newDiskUnlockCommand(),
+	)
+	return diskCommand
+}
+
+func newDiskCreateCommand() *cobra.Command {
+	var diskCreateCommand = &cobra.Command{
+		Use: "create DISK",
+		Example: `
+To create a new disk:
+$ limactl disk create DISK --size SIZE
+`,
+		Short: "Create a Lima disk",
+		Args:  cobra.ExactArgs(1),
+		RunE:  diskCreateAction,
+	}
+	diskCreateCommand.Flags().String("size", "", "configure the disk size")
+	diskCreateCommand.MarkFlagRequired("size")
+	return diskCreateCommand
+}
+
+func diskCreateAction(cmd *cobra.Command, args []string) error {
+	size, err := cmd.Flags().GetString("size")
+	if err != nil {
+		return err
+	}
+
+	diskSize, err := units.RAMInBytes(size)
+	if err != nil {
+		return err
+	}
+
+	// only exactly one arg is allowed
+	name := args[0]
+
+	diskDir, err := store.DiskDir(name)
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(diskDir); !errors.Is(err, fs.ErrNotExist) {
+		return fmt.Errorf("disk %q already exists (%q)", name, diskDir)
+	}
+
+	logrus.Infof("Creating a disk %q", name)
+
+	if err := os.MkdirAll(diskDir, 0700); err != nil {
+		return err
+	}
+
+	if err := qemu.CreateDataDisk(diskDir, int(diskSize)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func newDiskListCommand() *cobra.Command {
+	var diskListCommand = &cobra.Command{
+		Use: "list",
+		Example: `
+To list existing disks:
+$ limactl disk list
+`,
+		Short:   "List existing Lima disks",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
```

Comment:
Thought for a later PR (low priority):

I think it would be good to allow disk names to be passed as args (for symmetry with `limactl ls`, which allows instance names), and then only emit the data for those disk images.